editoast/src/valkey_utils.rs

```diff
@@ -202,19 +202,26 @@ pub enum ValkeyClient {
     NoCache,
 }
 
+pub struct ValkeyConfig {
+    /// Disables caching. This should not be used in production.
+    pub no_cache: bool,
+    pub is_cluster_client: bool,
+    pub valkey_url: Url,
+}
+
 impl ValkeyClient {
-    pub fn new(valkey_config: ValkeyConfig) -> Result<ValkeyClient> {
+    pub fn new(valkey_config: &ValkeyConfig) -> Result<ValkeyClient> {
```

Comment:
I made it a reference because there was no reason to consume the config since none of its fields needed to be moved. I removed the `as_str` as suggested, which consumes the URL, so I removed the reference as well. Cf. [`3894923` (#9605)](https://github.com/OpenRailAssociation/osrd/pull/9605/commits/38949235d3746a6fae496d37ddfac81a98e89f34)

Reply:
My remark was more on the side "yes, we technically don't need to consume it... but are we going to do with it after that anyway?". So it's a more logical reasoning than a technical one. Anyway, thanks for making the change 😄